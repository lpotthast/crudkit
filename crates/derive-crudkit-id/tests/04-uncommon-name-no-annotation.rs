#![allow(dead_code)]

use derive_crud_id::CrudId;

#[derive(CrudId)]
pub struct Foo {
    foo: i32,
}

fn main() {}
