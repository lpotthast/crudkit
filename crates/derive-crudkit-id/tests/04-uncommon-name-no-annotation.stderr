error: To derive CrudId, at least one id field must exist.

         = help: A field is an id field if it is (a) named "id" or (b) annotated with `#[crud_id(id)]`, both marking the field as part of the entities id. Specify id fields or remove the derive, if no id fields can be defined for this entity.

 --> tests/04-uncommon-name-no-annotation.rs:5:10
  |
5 | #[derive(CrudId)]
  |          ^^^^^^
  |
  = note: this error originates in the derive macro `CrudId` (in Nightly builds, run with -Z macro-backtrace for more info)
